runtime/onert/core/src/backend/cpu_common/DynamicTensorManager.cc

```diff
@@ -115,21 +115,27 @@ void DynamicTensorManager::deallocInput(ir::OperationIndex op_ind)
   auto &input_set = find->second;
   for (auto input_ind : input_set)
   {
-    if (!_tensors->getManagedTensor(input_ind)->is_dynamic())
+    auto *tensor = _tensors->getManagedTensor(input_ind).get();
+    if (!tensor->is_dynamic())
       continue;
 
     _dynamic_mem_mgr->deallocate(input_ind);
+    tensor->resetBuffer();
```

Comment:
Is this necessary here?

Reply:
> 3. After `deallocInput()` and `deallocSubghraphOutput()`, input/output tensors are explicitly marked as "the tensor does not have memory allocated". 
>    - Previously, the tensor might have `_allocator` even after `dealloc()`, and `Tensor::allocator != nullptr`, which is not our expectation.

This may be the reason? Not sure though😅

Reply:
`deallocate(..)` eventually releases calls `allocator.release()` which resets `_base` of `allocator`.
but this does not mean that `tensor->allocacator != nullptr`.
This could make some possible error later, e.g., in `Tensor.cc`, we check if tensor has memory or not by calling, e.g., `assert(tensor->allocator == nullptr)` but such call is currently not right after calling `deallocate(..)`